packages/swirl-components/src/components/swirl-popover-trigger/swirl-popover-trigger.tsx

```diff
@@ -99,12 +103,29 @@ export class SwirlPopoverTrigger {
 
   private onMouseleave = () => {
     if (!this.triggerOnHover) return;
-
     const popoverEl = this.getPopoverEl();
 
-    popoverEl.close(true);
+    popoverEl.addEventListener("mouseenter", () => {
```

Comment:
Issue: This will create a new event listener on each mouse leave event. The listeners for "mouseenter" and "mouseleave" should only be added once when a popoverEl is found, probably outside of this onMouseleave method. And you need to make sure to remove the event listeners when the popoverEl is changes or the component is destroyed.

Reply:
Because one popover can have multiple trigger elements i can't have a permanent listener attached. I solved it now by making the event listeners only invoke 'once'.

Reply:
But a trigger can only have one popover. Multiple triggers adding a mouseenter listener to the popover should be fine.

The once solution still produces listeners that are never removed, unfortunately. For each mouseleave on the trigger you register a mouseenter on the popover. There is no guarantee that the mouseenter on the popover ever happens  and is then unregistered.

Reply:
yeah the mouseenter event was fine, but on leave the second trigger element would close the popover. I resolved it now by keeping track of which popover trigger has caused the popover to open, have another look :)